app/charts/pie/pie-state.tsx

```diff
@@ -100,51 +73,64 @@ const usePieState = (
   const getX = useSegment(fields.segment.componentIri);
 
   // Sort data
-  const sortingType = fields.segment.sorting?.sortingType;
   const sortingOrder = fields.segment.sorting?.sortingOrder;
 
   // Data actually sorted in pie(),
-  // Sorting here only useful to legend items.
-  const sortedData = useMemo(() => {
-    return sortData({ data, sortingType, sortingOrder, getX, getY });
-  }, [data, getX, getY, sortingType, sortingOrder]);
-
-  const plottableSortedData = usePlottableData({
-    data: sortedData,
+  const plottableData = usePlottableData({
+    data: data,
     plotters: [getY],
   });
 
   // Apply end-user-activated interactive filters to the stack
   const preparedData = usePreparedData({
     legendFilterActive: interactiveFiltersConfig?.legend.active,
-    sortedData: plottableSortedData,
+    sortedData: data,
     interactiveFilters,
     getSegment: getX,
   });
 
-  const { segmentValuesByValue, segmentDimension } = useMemo(() => {
-    const segmentDimension = dimensions.find(
-      (d) => d.iri === fields.segment?.componentIri
-    ) as $FixMe;
-    return {
-      segmentDimension,
-      segmentValuesByValue: keyBy(segmentDimension.values, (x) => x.value),
-      segmentValuesByLabel: keyBy(segmentDimension.values, (x) => x.label),
-    };
-  }, [dimensions, fields.segment?.componentIri]);
+  const { segmentValuesByValue, segmentDimension, segmentValuesByLabel } =
+    useMemo(() => {
+      const segmentDimension = dimensions.find(
+        (d) => d.iri === fields.segment?.componentIri
+      ) as $FixMe;
+      return {
+        segmentDimension,
+        segmentValuesByValue: keyBy(
+          segmentDimension.values as DimensionValue[] as DimensionValue[],
+          (x) => x.value
+        ),
+        segmentValuesByLabel: keyBy(
+          segmentDimension.values as DimensionValue[],
+          (x) => x.label
+        ),
+      };
+    }, [dimensions, fields.segment?.componentIri]);
 
   // Map ordered segments to colors
   const colors = useMemo(() => {
     const colors = scaleOrdinal<string, string>();
-    const segments = Array.from(
-      new Set(plottableSortedData.map((d) => getX(d)))
+    const measureBySegment = Object.fromEntries(
+      plottableData.map((d) => [getX(d), getY(d)])
+    );
+    const uniqueSegments = Object.entries(measureBySegment)
+      .filter((x) => x[1])
```

Comment:
`.filter(x[1] !== null && x[1] !== undefined)`? With the current logic it would filter out 0s right? Or was this intended, as anyway we can't see 0s on pie chart (but on the other hand we still we could see it in a legend)?

Reply:
Good point :+1: